### ISSUE TITLE
pyright's `collections.namedtuple` model erroneously synthesizes `keys` and `items` methods

### ISSUE THREAD

**Issue by Sachaa-Thanasius** [microsoft/pyright#7768]
**Describe the bug**
When defining a namedtuple `A` (or NamedTuple with the functional call) with or without `items` and `keys` as field names, an instance of `A` will have `items` and `keys` methods that appear to return `list[str]`. As far as I know, namedtuples have never had those methods, but maybe they did and don't anymore for python3.8+.

**Code or Screenshots**
```python
from collections import namedtuple
from typing import NamedTuple
from typing_extensions import reveal_type


Example0 = namedtuple("Example0", ["name", "items"])
ex0 = Example0(True, (1, 2))
reveal_type(ex0.name)  # Type of "ex0.name" is "Unknown"
reveal_type(ex0.keys)  # Type of "ex0.keys" is "() -> list[str]"
reveal_type(ex0.items)  # Type of "ex0.items" is "() -> list[str]"

Example1 = NamedTuple("Example1", [("name", bool)])
ex1 = Example1(True)
reveal_type(ex1.name)  # Type of "ex1.name" is "bool"
reveal_type(ex1.keys)  # Type of "ex1.keys" is "() -> list[str]"
reveal_type(ex1.items)  # Type of "ex1.items" is "() -> list[str]"
```

**VS Code extension or command-line**
I've noticed this with Pylance in VS Code as well as pyright playground (version 1.1.360).

**Extra information**
I tracked down the synthesis to these lines:

https://github.com/microsoft/pyright/blob/4bc8428d756ae09dbb5ba9eb5a8ec12b3aded108/packages/pyright-internal/src/analyzer/namedTuples.ts#L385-L392

Fixing this seems as simple as deleting them, but I'm not certain.

**Comment by erictraut** [comment]
Hmm, that code has been in pyright for a long time, and I don't recall why I originally added it.

It appears that `item` is present in the case of `Example0` but not in `Example1`, and `keys` isn't present on either.

**Comment by Sachaa-Thanasius** [comment]
Yeah, I was just demonstrating that `Example0` will have `items` as an "attribute" at runtime due to me putting it in the functional constructor, but the annotation for it is still incorrect. No matter what, the synthesis of `keys` and `items` methods seems to occur, hardcoded with `() -> list[str]` signatures.

**Comment by erictraut** [comment]
Ah, I missed the fact that you added an item named `items` in `Example0`. Thanks for pointing that out.

**Comment by Sachaa-Thanasius** [comment]
Out of curiosity, did you find out why items and keys were synthesized there in the first place?

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.361, which I just published.
